py3status/modules/mpris.py

```diff
@@ -277,40 +281,67 @@ def _get_text(self, i3s_config):
-                         'action':    'Pause'},
-            'play':     {'clickable': 'CanPlay',
-                         'icon':      self.icon_play,
-                         'action':    'Play'},
-            'stop':     {'clickable': 'True',
-                         'icon':      self.icon_stop,
-                         'action':    'Stop'},
-            'next':     {'clickable': 'CanGoNext',
-                         'icon':      self.icon_next,
-                         'action':    'Next'},
-            'previous': {'clickable': 'CanGoPrevious',
-                         'icon':      self.icon_previous,
-                         'action':    'Previous'}
+            'pause':    {'action':    'Pause',
+                         'clickable': 'CanPause',
+                         'enable':    None,
+                         'icon':      self.icon_pause},
+            'play':     {'action':    'Play',
+                         'clickable': 'CanPlay',
+                         'enable':    None,
+                         'icon':      self.icon_play},
+            'stop':     {'action':    'Stop',
+                         'clickable': 'True',
+                         'enable':    None,
+                         'icon':      self.icon_stop},
+            'next':     {'action':    'Next',
+                         'clickable': 'CanGoNext',
+                         'enable':    None,
+                         'icon':      self.icon_next},
+            'previous': {'action':    'Previous',
+                         'clickable': 'CanGoPrevious',
+                         'enable':    None,
+                         'icon':      self.icon_previous}
         }
 
         state = 'pause' if self._player.PlaybackStatus == 'Playing' else 'play'
         control_states['toggle'] = control_states[state]
 
         return control_states
 
-    def _get_response_buttons(self):
+    def _get_button_state(self, control_state):
+        playback_status = self._player.PlaybackStatus
+        clickable = getattr(self._player, control_state['clickable'], True)
+
+        if control_state['action'] == 'Play' and playback_status == 'Playing':
+            clickable = False
+        elif control_state['action'] == 'Pause' and (
+             playback_status == 'Stopped' or playback_status == 'Paused'):
+            clickable = False
+        elif control_state['action'] == 'Stop' and playback_status == 'Stopped':
+            clickable = False
+
+        return clickable
+
+    def _get_response_buttons(self, i3s_config):
         response = []
         buttons_order = self.buttons_order.split(',')
         available = self._control_states.keys()
 
         for button in [e for e in buttons_order if e in available]:
             control_state = self._control_states[button]
-            clickable = getattr(self._player, control_state['clickable'], True)
+            control_state['enable'] = self._get_button_state(control_state)
```

Comment:
I think this is a backward step a it makes control_states have state, which feels wrong to me.  There isn't a need for this.  You are only really needing it here.  You do use it in `on_click()` but then you are using a cached version and it has no real value.  In the on_click if you care then why not just recalculate the value?